app/com/yetu/oauth2provider/controllers/OAuth2ResourceServer.scala

```diff
@@ -1,74 +1,90 @@
     implicit request =>
+      authorize(authorizationHandler) { authInfo =>
 
-      Logger.debug(s"request = $request \n \t body = ${request.body}")
-      authorize(authorizationHandler) {
-        authInfo =>
-          Logger.debug(s"request has been authorized... = $request \n \t body = ${request.body}")
+        val user = authInfo.user
 
-          // current user is defined via Nulab's library and AuthorizationHandler
-          val user = authInfo.user
+        Logger.debug(s"request has been authorized...user = $user")
+        Logger.debug(s"request has been authorized...scope = ${authInfo.scope}")
+        Logger.debug(s"request has been authorized...authInfo = $authInfo")
 
-          Logger.debug(s"request has been authorized...user = $user")
-          Logger.debug(s"request has been authorized...scope = ${authInfo.scope}")
-          Logger.debug(s"request has been authorized...authInfo = $authInfo")
+        val outputJson: Option[JsValue] = scopeService.getInfoByScope(
+          authInfo.user,
+          authInfo.scope.getOrElse(Config.SCOPE_ID)).map(user => Json.toJson(user))
 
-          val outputJson: Option[JsValue] = scopeService.getInfoByScope(authInfo.user, authInfo.scope.getOrElse(Config.SCOPE_ID)).map(user => Json.toJson(user))
+        val result = outputJson match {
+          case Some(userJson) => Ok(userJson)
+          case None => BadRequest("You have been authorized, but your scope is invalid " +
+            "or not authorized, you cannot access any data for this user.")
+        }
 
-          Logger.debug(s"request has been authorized...outputJson = $outputJson")
-          outputJson match {
-            case None           => Future.successful(BadRequest("You have been authorized, but your scope is invalid or not authorized, you cannot access any data for this user."))
-            case Some(userJson) => Future.successful(Ok(userJson))
-          }
+        Future.successful(result)
       }
   }
 
-  def updateUserProfile = Action.async(parse.json[DataUpdateRequest]) {
-    //TODO: use async interface properly already in the service class instead of a Future.successful()
+  def updateUserProfile() = Action.async(parse.json[DataUpdateRequest]) {
     implicit request =>
       authorize(authorizationHandler) {
         authInfo =>
           {
-            val result: Option[Result] = for {
-              scope: String <- authInfo.scope if scope == Config.SCOPE_CONTROLCENTER
-            } yield personService.updateUserProfile(authInfo.user, request.body)
 
-            result.fold(Future.successful(Unauthorized(s"This access token is valid but it does not have the authorized scope to update user contact and personal information."))
-            )(resultOfUpdate => Future.successful(resultOfUpdate))
+            if (authInfo.scope.getOrElse("").equals(Config.SCOPE_CONTROLCENTER)) {
+
+              val firstName = StringUtils.isFull(request.body.firstName) match {
```

Comment:
I don't usually pattern-match for boolean values, use if-else. Pattern matching in this case is overkill and inefficient.